eox_tenant/utils.py

```diff
@@ -19,6 +19,34 @@
     r'+[A-Za-z0-9][A-Za-z0-9-_]{0,61}'  # First 61 characters of the gTLD
     r'[A-Za-z]$'  # Last character of the gTLD
 )
+NON_SERIALIZABLE = object()
+
+
+def clean_serializable_values(obj):
+    """
+    Returns only serializable values
+
+    If an object is not json serializable returns NON_SERIALIZABLE.
+    In case of a list or dict ignore the NON_SERIALIZABLE values.
+    """
+    if isinstance(obj, (str, int, float, bool, type(None))):
+        return obj
+
+    if isinstance(obj, list):
+        return [
+            cleaned_item
+            for cleaned_item in (clean_serializable_values(item) for item in obj)
+            if cleaned_item is not NON_SERIALIZABLE
+        ]
+
+    if isinstance(obj, dict):
+        return {
+            key: cleaned_value
+            for key, cleaned_value in ((k, clean_serializable_values(v)) for k, v in obj.items())
+            if cleaned_value is not NON_SERIALIZABLE
+        }
+
+    return NON_SERIALIZABLE
```

Comment:
What happens when you return NON_SERIALIZABLE?
Will this fail the celery serialization? or is this a placeholder to signify that there was a key with some value, but the value would not be serialized and therefore is empty?

Reply:
Yes, I use `NON_SERIALIZABLE` as a sentinel to mark values that are not json serializable. When returning from the recursive call the non serializable values get cleaned from the nested data structures (`list`, `dict`). I was going to use `None` but `None` corresponds to `null` and I didn't want to delete keys with those values.

If you were to run something like.

```python
c = {
    "String": "str",
    "integer": 1,
    "bool": True,
    "none": None,
    "object": Exception,
    "list": [1, Exception],
    "dict": {1: "str", "object": Exception},
}
print(clean_serializable_values(c))
```

You would get:

```python
{
    'String': 'str',
    'integer': 1,
    'bool': True,
    'none': None,
    'list': [1],
    'dict': {1: 'str'}
}
```